includes/sanitizers/class-amp-gallery-block-sanitizer.php

```diff
@@ -149,105 +125,17 @@ public function sanitize() {
 				continue;
 			}
 
-			list( $width, $height ) = $this->get_carousel_dimensions( $node );
-
-			$amp_carousel = AMP_DOM_Utils::create_node(
-				$this->dom,
-				'amp-carousel',
-				[
-					'width'  => $width,
-					'height' => $height,
-					'type'   => 'slides',
-					'layout' => 'responsive',
-				]
-			);
-
+			$images_and_captions = [];
 			foreach ( $images as $image ) {
-				$slide = AMP_DOM_Utils::create_node(
-					$this->dom,
-					'div',
-					[ 'class' => 'slide' ]
-				);
-
-				// Ensure the image fills the entire <amp-carousel>, so the possible caption looks right.
-				if ( 'amp-img' === $image->tagName ) {
-					$image->setAttribute( 'layout', 'fill' );
-					$image->setAttribute( 'object-fit', 'cover' );
-				} elseif ( isset( $image->firstChild->tagName ) && 'amp-img' === $image->firstChild->tagName ) {
-					// If the <amp-img> is wrapped in an <a>.
-					$image->firstChild->setAttribute( 'layout', 'fill' );
-					$image->firstChild->setAttribute( 'object-fit', 'cover' );
-				}
-
-				$possible_caption_text = $this->possibly_get_caption_text( $image );
-				$slide->appendChild( $image );
-
-				// Wrap the caption in a <div> and <span>, and append it to the slide.
-				if ( $possible_caption_text ) {
-					$caption_wrapper = AMP_DOM_Utils::create_node(
-						$this->dom,
-						'div',
-						[ 'class' => 'amp-wp-gallery-caption' ]
-					);
-					$caption_span    = AMP_DOM_Utils::create_node( $this->dom, 'span', [] );
-					$text_node       = $this->dom->createTextNode( $possible_caption_text );
-
-					$caption_span->appendChild( $text_node );
-					$caption_wrapper->appendChild( $caption_span );
-					$slide->appendChild( $caption_wrapper );
-				}
-
-				$amp_carousel->appendChild( $slide );
+				$images_and_captions[] = [ $image, $this->possibly_get_caption_text( $image ) ];
 			}
 
-			$gallery_node->parentNode->replaceChild( $amp_carousel, $gallery_node );
-		}
-		$this->did_convert_elements = true;
-	}
+			$amp_carousel      = new AMP_Carousel( $this->dom );
+			$amp_carousel_node = $amp_carousel->create_and_get( $images_and_captions );
 
-	/**
-	 * Get carousel height by containing images.
-	 *
-	 * @param DOMElement $element The UL element.
-	 * @return array {
-	 *     Dimensions.
-	 *
-	 *     @type int $width  Width.
-	 *     @type int $height Height.
-	 * }
-	 */
-	protected function get_carousel_dimensions( $element ) {
```

Comment:
This is mainly moved to [AMP_Carousel](https://github.com/ampproject/amp-wp/pull/3659/files#diff-1892bee5b65cec2225d075857448d046R124), with few changes.